src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -108,76 +179,72 @@ def __call__(self, images=None, task_inputs=None, segmentation_maps=None, **kwar
             - **pixel_values** -- Pixel values to be fed to a model. Returned when `images` is not `None`.
         """
 
-        if task_inputs is None:
-            raise ValueError("You have to specify the task_input. Found None.")
-        elif images is None:
-            raise ValueError("You have to specify the image. Found None.")
-
-        if not all(task in ["semantic", "instance", "panoptic"] for task in task_inputs):
-            raise ValueError("task_inputs must be semantic, instance, or panoptic.")
-
-        encoded_inputs = self.image_processor(images, task_inputs, segmentation_maps, **kwargs)
-
+        output_kwargs = self._merge_kwargs(
+            OneFormerProcessorKwargs,
+            tokenizer_init_kwargs=self.tokenizer.init_kwargs,
+            **kwargs,
+            **self.prepare_and_validate_optional_call_args(*args),
+        )
+        task_inputs = output_kwargs["images_kwargs"].pop("task_inputs", None)
+        segmentation_maps = output_kwargs["images_kwargs"].pop("segmentation_maps", None)
         if isinstance(task_inputs, str):
             task_inputs = [task_inputs]
+        self._validate_input_types(images, task_inputs)
 
-        if isinstance(task_inputs, List) and all(isinstance(task_input, str) for task_input in task_inputs):
-            task_token_inputs = []
-            for task in task_inputs:
-                task_input = f"the task is {task}"
-                task_token_inputs.append(task_input)
-            encoded_inputs["task_inputs"] = self._preprocess_text(task_token_inputs, max_length=self.task_seq_length)
-        else:
-            raise TypeError("Task Inputs should be a string or a list of strings.")
+        encoded_inputs = self.image_processor(images, task_inputs, segmentation_maps, **output_kwargs["images_kwargs"])
 
-        if hasattr(encoded_inputs, "text_inputs"):
-            texts_list = encoded_inputs.text_inputs
-
-            text_inputs = []
-            for texts in texts_list:
-                text_input_list = self._preprocess_text(texts, max_length=self.max_seq_length)
-                text_inputs.append(text_input_list.unsqueeze(0))
+        task_token_inputs = []
+        for task in task_inputs:
+            task_input = f"the task is {task}"
+            task_token_inputs.append(task_input)
+        encoded_inputs["task_inputs"] = self._preprocess_text(
+            task_token_inputs,
+            max_length=self.task_seq_length,
+            text_kwargs=output_kwargs["text_kwargs"],
+        )
 
+        if hasattr(encoded_inputs, "text_inputs"):
+            text_inputs = [
+                self._preprocess_text(
+                    texts,
+                    max_length=self.max_seq_length,
+                    text_kwargs=output_kwargs["text_kwargs"],
+                ).unsqueeze(0)
+                for texts in encoded_inputs.text_inputs
```

Comment:
let's od a for loop it's more redable. Also this is now almost exactly the same as `encode_inputs` why don't we call that funciton instead / merge both?